Tests/Indicators/CompositeIndicatorTests.cs

```diff
@@ -93,5 +93,43 @@ public void CallsDelegateCorrectly()
             Assert.AreEqual(left.PeriodsSinceMaximum, 0);
             Assert.AreEqual(right.PeriodsSinceMinimum, 0);
         }
+
+        [Test]
+        public void TimeAdvancesWithDataToBothSides()
+        {
+            var left = new Maximum("left", 2);
+            var right = new Minimum("right", 2);
+            var composite = new CompositeIndicator<IndicatorDataPoint>(left, right, (l, r) => l.Current.Value + r.Current.Value);
+
+            Assert.AreEqual(DateTime.MinValue, composite.Current.Time);
+
+            left.Update(DateTime.Today, 1m);
+            right.Update(DateTime.Today, -1m);
+
+            Assert.AreEqual(DateTime.Today, composite.Current.Time);
+
+            left.Update(DateTime.Today.AddDays(1), -1m);
+            right.Update(DateTime.Today.AddDays(1), 1m);
+
+            Assert.AreEqual(DateTime.Today.AddDays(1), composite.Current.Time);
```

Comment:
Leaving a comment for posterity: usually we'll want to save the `DateTime.Today/Now/UtcNow` in a variable to ensure that time doesn't change on us while we're running our tests. 

In this case, it's unlikely that the test will fail, but it would fail at midnight if it was ran then :)

Reply:
Gotcha, better safe than sorry 😁